src/PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.cpp

```diff
@@ -0,0 +1,249 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/GrMhd/BondiHoyleAccretion.hpp"
+
+#include <cmath>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/// \cond
+namespace grmhd {
+namespace AnalyticData {
+
+BondiHoyleAccretion::BondiHoyleAccretion(
+    const double black_hole_mass, const double black_hole_spin,
+    const double rest_mass_density, const double flow_speed,
+    const double magnetic_field_strength, const double polytropic_constant,
+    const double polytropic_exponent) noexcept
+    : black_hole_mass_(black_hole_mass),
+      black_hole_spin_(black_hole_spin),
+      rest_mass_density_(rest_mass_density),
+      flow_speed_(flow_speed),
+      magnetic_field_strength_(magnetic_field_strength),
+      polytropic_constant_(polytropic_constant),
+      polytropic_exponent_(polytropic_exponent),
+      equation_of_state_{polytropic_constant_, polytropic_exponent_},
+      kerr_schild_coords_{black_hole_spin_} {}
+
+void BondiHoyleAccretion::pup(PUP::er& p) noexcept {
+  p | black_hole_mass_;
+  p | black_hole_spin_;
+  p | rest_mass_density_;
+  p | flow_speed_;
+  p | magnetic_field_strength_;
+  p | polytropic_constant_;
+  p | polytropic_exponent_;
+  p | equation_of_state_;
+  p | kerr_schild_coords_;
+}
+
+template <typename DataType>
+typename hydro::Tags::SpatialVelocity<DataType, 3>::type
+BondiHoyleAccretion::spatial_velocity(const DataType& r_squared,
+                                      const DataType& sin_theta_squared) const
+    noexcept {
+  auto result =
+      make_with_value<typename hydro::Tags::SpatialVelocity<DataType, 3>::type>(
+          r_squared, 0.0);
+
+  const DataType cos_theta_squared = 1.0 - sin_theta_squared;
+  const DataType sigma =
+      r_squared + square(black_hole_spin_) * cos_theta_squared;
+
+  result.get(0) = flow_speed_ * sqrt(cos_theta_squared) /
```

Comment:
get<0>(result) = ...
I think this is preferred because it's a compile-time get instead of a run-time get?